### PR TITLE
[DNM] divide localstate into raftstate and applystate

### PR DESCRIPTION
**Pull request by BusyJay**
ref https://github.com/pingcap/tikv/pull/785

@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -37,9 +37,12 @@ message StoreIdent {
 
 message RaftLocalState {
     optional raftpb.HardState hard_state        = 1;
-    optional uint64 applied_index               = 2;
-    optional uint64 last_index                  = 3;
-    optional RaftTruncatedState truncated_state = 4;
+    optional uint64 last_index                  = 2;
+}
+
+message ApplyState {
```
Comment (ngaut) [inline proto/raft_serverpb.proto]: s/ApplyState/RaftApplyState/

**Comment by siddontang** [comment]
Naming RaftLocalState is proper now?

**Comment by BusyJay** [comment]
Yes, both last index and hard state are raft related fields.

**Comment by BusyJay** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**